NetCoreRestApi/DataLayer/Converters/ProductEntityConverter.cs

```diff
@@ -0,0 +1,50 @@
+﻿using DataLayer.Models;
+using Common.Converter;
+using DataLayer.Entities;
+using System.Linq;
+using System.Collections.Generic;
+
+namespace DataLayer.Converters
+{
+    public class ProductEntityConverter : IConverter<ProductEntity, ProductModel>
+    {
+        private readonly CategoryEntityConverter _categoryEntityConverter = new CategoryEntityConverter();
+
+        public ProductModel ConvertFrom(ProductEntity productEntity)
+        {
+            if (productEntity == null)
+            {
+                return null;
+            }
+            var productModel = new ProductModel
+            {
+                ID = productEntity.ID,
+                Name = productEntity.Name,
+                Description = productEntity.Description,
+                CategoryList = new List<CategoryModel>(from category in productEntity.Categories 
```

Comment:
Create separated method for converting categories. Use extension methods instead of from..in ...

Reply:
Ok, I will try that

Reply:
done